healthcareai/common/model_eval.py

```diff
@@ -114,34 +172,105 @@ def calculate_regression_metrics(trained_sklearn_estimator, x_test, y_test):
-        trained_sklearn_estimator (sklearn.base.BaseEstimator): a scikit-learn estimator that has been `.fit()`
+        classes (list): A list of classes.
+
+    Returns:
+        object: The likely positive classes.
+    """
+    if len(classes) != 2:
+        raise HealthcareAIError(
+            'Guessing the positive label is meant for binary tasks only.'
+            ' You passed in {}'.format(classes))
+
+    result = list(classes)
+    result.sort(reverse=True)
+
+    return result[0]
+
+
+def compute_classification_metrics(
+        trained_sklearn_estimator,
+        x_test,
+        y_test,
+        binary_positive_label=None):
+    """
+    Compute classification metrics for a trained estimator.
+
+    Note this computes accuracy and a confusion matrix and optionally adds PR
+    and ROC for binary classification tasks.
+
+    Args:
+        trained_sklearn_estimator (sklearn.base.BaseEstimator): a scikit-learn
+        estimator that has been `.fit()`
         x_test (numpy.ndarray): A 2d numpy array of the x_test set (features)
         y_test (numpy.ndarray): A 1d numpy array of the y_test set (predictions)
+        binary_positive_label (int|str): Optional positive class label for binary classification tasks.
 
     Returns:
         dict: A dictionary of metrics objects
     """
-    # Squeeze down y_test to 1D
+    # Squeeze down y_test from nD to 1D
     y_test = np.squeeze(y_test)
 
     _validate_predictions_and_labels_are_equal_length(x_test, y_test)
 
     # Get binary and probability classification predictions
-    binary_predictions = np.squeeze(trained_sklearn_estimator.predict(x_test))
-    probability_predictions = np.squeeze(trained_sklearn_estimator.predict_proba(x_test)[:, 1])
+    class_predictions = np.squeeze(trained_sklearn_estimator.predict(x_test))
+    raw_predictions = trained_sklearn_estimator.predict_proba(x_test)
+    probability_predictions = np.squeeze(raw_predictions[:, 1])
 
-    # Calculate accuracy
-    accuracy = skmetrics.accuracy_score(y_test, binary_predictions)
-    roc = compute_roc(y_test, probability_predictions)
-    pr = compute_pr(y_test, probability_predictions)
+    # Compute the accuracy
+    accuracy = skmetrics.accuracy_score(y_test, class_predictions)
 
-    # Unpack the roc and pr dictionaries so the metric lookup is easier for plot and ensemble methods
-    return {'accuracy': accuracy, **roc, **pr}
+    # Compute the confusion matrix
+    confusion_matrix, _ = compute_confusion_matrix(y_test, class_predictions)
+
+    results = {'accuracy': accuracy, 'confusion_matrix': confusion_matrix}
+
+    # Select appropriate metric for binary vs multi-class classifications
+    number_classes = len(np.unique(y_test))
+    if number_classes == 2:
+
+        if binary_positive_label is None:
+            # Guess the positive class if none is specified
+            positive_label = guess_positive_label(y_test.unique())
+        else:
+            positive_label = binary_positive_label
+
```

Comment:
Might need to adjust the `probability_predictions` here if the user specifies the `binary_positive_label`

Reply:
Nice. Testing now.